go.mod

```diff
@@ -3,89 +3,103 @@ module k8s.io/node-problem-detector
 go 1.20
 
 require (
-	cloud.google.com/go v0.45.1
+	cloud.google.com/go/compute/metadata v0.2.3
 	code.cloudfoundry.org/clock v0.0.0-20180518195852-02e53af36e6c
 	contrib.go.opencensus.io/exporter/prometheus v0.0.0-20190427222117-f6cda26f80a3
 	contrib.go.opencensus.io/exporter/stackdriver v0.13.4
 	github.com/acobaugh/osrelease v0.0.0-20181218015638-a93a0a55a249
 	github.com/avast/retry-go v2.4.1+incompatible
 	github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e
 	github.com/euank/go-kmsg-parser v2.0.0+incompatible
-	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
+	github.com/golang/glog v1.1.0
 	github.com/google/cadvisor v0.36.0
 	github.com/hpcloud/tail v1.0.0
 	github.com/onsi/ginkgo v1.16.5
-	github.com/onsi/gomega v1.10.1
+	github.com/onsi/gomega v1.27.6
 	github.com/pborman/uuid v1.2.0
-	github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4
-	github.com/prometheus/common v0.4.1
-	github.com/prometheus/procfs v0.2.0
+	github.com/prometheus/client_model v0.3.0
+	github.com/prometheus/common v0.37.0
+	github.com/prometheus/procfs v0.8.0
 	github.com/shirou/gopsutil v2.19.12+incompatible
 	github.com/spf13/pflag v1.0.5
-	github.com/stretchr/testify v1.6.1
-	go.opencensus.io v0.22.4
-	golang.org/x/oauth2 v0.0.0-20190604053449-0f29369cfe45
-	golang.org/x/sys v0.0.0-20210112080510-489259a85091
-	google.golang.org/api v0.10.0
-	k8s.io/api v0.0.0-20190816222004-e3a6b8045b0b
-	k8s.io/apimachinery v0.0.0-20190816221834-a9f1d8a9c101
+	github.com/stretchr/testify v1.8.1
+	go.opencensus.io v0.24.0
+	golang.org/x/crypto v0.1.0
+	golang.org/x/oauth2 v0.7.0
+	golang.org/x/sys v0.8.0
+	google.golang.org/api v0.114.0
+	k8s.io/api v0.17.2
+	k8s.io/apimachinery v0.17.2
 	k8s.io/client-go v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible
-	k8s.io/kubernetes v1.14.6
```

Comment:
Why are we removing kubernetes as a dep?

Reply:
In the later version, we need to add a lot of replace statements. It is much simpler if there are fewer dependencies.

Reply:
I understand, but not sure if this is correct. Let's keep it on the same version for now, we can send a follow-up PR just to update it.

Reply:
ssh packageis the only package